### PR TITLE
some const cleanup: remove unnecessary attributes, add const-hack indications

### PR DESCRIPTION
**Pull request by RalfJung**
I learned that we use `FIXME(const-hack)` on top of the "const-hack" label. That seems much better since it marks the right place in the code and moves around with the code. So I went through the PRs with that label and added appropriate FIXMEs in the code. IMO this means we can then remove the label -- Cc @rust-lang/wg-const-eval.

I also noticed some const stability attributes that don't do anything useful, and removed them.

r? @fee1-dead

### CODE REVIEW

**Comment by RalfJung** [comment]
Hm, I guess I should add a tracking issue for const_slice_index instead of removing it entirely...

**Comment by RalfJung** [comment]
No wait, this is entirely private functions and trait impls. It can't be doing anything.

File: compiler/rustc_mir_transform/src/pass_manager.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ fn to_profiler_name(type_name: &'static str) -> &'static str {
 
 // const wrapper for `if let Some((_, tail)) = name.rsplit_once(':') { tail } else { name }`
 const fn c_name(name: &'static str) -> &'static str {
-    // FIXME Simplify the implementation once more `str` methods get const-stable.
+    // FIXME(const-hacks) Simplify the implementation once more `str` methods get const-stable.
```
Comment (PatchMixolydic) [inline compiler/rustc_mir_transform/src/pass_manager.rs:45]: Minor typo (in this file only):
```suggestion
    // FIXME(const-hack): Simplify the implementation once more `str` methods get const-stable.
```

File: compiler/rustc_mir_transform/src/pass_manager.rs [hunk 2]
```diff
@@ -61,7 +61,7 @@ const fn c_name(name: &'static str) -> &'static str {
 /// loop that goes over each available MIR and applies `run_pass`.
 pub(super) trait MirPass<'tcx> {
     fn name(&self) -> &'static str {
-        // FIXME Simplify the implementation once more `str` methods get const-stable.
+        // FIXME(const-hacks) Simplify the implementation once more `str` methods get const-stable.
```
Comment (PatchMixolydic) [inline compiler/rustc_mir_transform/src/pass_manager.rs:64]:
```suggestion
        // FIXME(const-hack): Simplify the implementation once more `str` methods get const-stable.
```

File: compiler/rustc_mir_transform/src/pass_manager.rs [hunk 3]
```diff
@@ -89,7 +89,7 @@ pub(super) trait MirPass<'tcx> {
 /// disabled (via the `Lint` adapter).
 pub(super) trait MirLint<'tcx> {
     fn name(&self) -> &'static str {
-        // FIXME Simplify the implementation once more `str` methods get const-stable.
+        // FIXME(const-hacks) Simplify the implementation once more `str` methods get const-stable.
```
Comment (PatchMixolydic) [inline compiler/rustc_mir_transform/src/pass_manager.rs:92]:
```suggestion
        // FIXME(const-hack): Simplify the implementation once more `str` methods get const-stable.
```

**Review by fee1-dead — APPROVED** [review]

**Pull request merged**